### ISSUE TITLE
IDISP004 should not warn on disposed instance

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#79]
```cs
namespace RoslynSandbox
{
    using System.IO;

    class Foo
    {
        public Foo()
        {
            var text = Disposed().ToString();
        }

        private static object Disposed()
        {
            using (var stream = File.OpenRead(string.Empty))
            {
#pragma warning disable IDISP011 // Don't return disposed instance.
                return stream;
#pragma warning restore IDISP011 // Don't return disposed instance.
            }
        }
    }
}
```

No a very useful edge case.
